adapt/tools/text/trie.py

```diff
@@ -102,31 +216,114 @@ def remove(self, iterable, data=None, index=0):
             else:
                 return False
         elif iterable[index] in self.children:
-            return self.children[iterable[index]].remove(iterable, index=index+1, data=data)
+            return self.children[iterable[index]].remove(
+                iterable, index=index + 1, data=data)
         else:
             return False
 
 
 class Trie(object):
+    """Interface for the tree"""
+
     def __init__(self, max_edit_distance=0, match_threshold=0.0):
+        """Init the Trie object and create root node.
+
+        Creates an Trie object with a root node with the passed in
+        max_edit_distance and match_threshold.
+
+        Parameters
+        ----------
+        max_edit_distance: int
+        match_threshold: int
+
+        Notes
+        -----
+        This never seems to get called with max_edit_distance or match_threshold
+        """
         self.root = TrieNode('root')
         self.max_edit_distance = max_edit_distance
         self.match_threshold = match_threshold
 
+    def __str__(self):
+        """Use to get a string represntation of the Trie from Root Node
+
+        Returns
+        -------
+        string - A String representation of the entire Trie
+        """
+        return str("Trie Object{%s}" % str(self.root))
+
+    def json(self):
+        """Created to show the tree"""
+        return self.root.json()
+
     def gather(self, iterable):
+        """Calls the lookup with gather True Passing iterable and yields
+        the result.
+        """
         for result in self.lookup(iterable, gather=True):
             yield result
 
+    def checkForMissingEntites(self, entities):
+        """Used for checking a given list for missing Entities
+
+        Parameters
+        ----------
+        entities - []/string - representing entities to search for
+
+        Returns
+        -------
+        [] - a list of the missing entities
+        Will be None if all the entites are found
+        """
+        Trie_entites = self.root.entities()
```

Comment:
it would probably be more efficient to track entities on the Trie object (as opposed to the Node object) in a set, and then use set difference. Most of this function goes away.